### ISSUE TITLE
Unable to create symlink

### ISSUE THREAD

**Issue by nikitashalnov** [voxpupuli/puppet-splunk#212]
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: 1.10.14-1stretch (for example)
- Distribution: Debian 9
- Module version: 7.3.0

## Description
Splunk versions above 7.2.1 don't create file /etc/init.d/splunk during the command `splunk enable boot-start -user ${splunk_user}`. It creates symlink on systemd unit (service):

```bash
ls -l /etc/systemd/system/multi-user.target.wants/SplunkForwarder.service 
lrwxrwxrwx 1 root root 43 Jan 16 13:30 /etc/systemd/system/multi-user.target.wants/SplunkForwarder.service -> /etc/systemd/system/SplunkForwarder.service
```

But in `manifests/platform/posix.pp` execs expect that file `/etc/init.d/splunk` is created. So on every run of puppet agent, agent tries to create this symlink and gets an error:

```bash
Notice: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]/returns: Unable to create symlink='/etc/systemd/system/multi-user.target.wants/SplunkForwarder.service'
Notice: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]/returns: : File exists
Error: '/opt/splunkforwarder/bin/splunk enable boot-start -user root' returned 8 instead of one of [0]
Error: /Stage[main]/Splunk::Platform::Posix/Exec[enable_splunkforwarder]/returns: change from notrun to 0 failed: '/opt/splunkforwarder/bin/splunk enable boot-start -user root' returned 8 instead of one of [0]
```

I [commited](https://github.com/nikitashalnov/puppet-splunk/commit/eb6b794f29c7843d23b062541c32cecd2f87ef0f) some changes to my fork, but not sure that this is the proper way to fix this problem.

**Comment by HernaldoM** [comment]
hi @nikitashalnov any update on when this could be completed? we are wanting to run 7.2.3 but aren't able to because of this issue.

Thanks

**Comment by mwhitake** [comment]
I ran into the same problem. I was able to fix that by adding a variable and checking for the presence of the systemd start script on EL7 hosts.  
However, once I had that issue resolved, it revealed a different issue.  The splunk boot-enable command creates the systemd start scripts with mixed case names, and gives the forwarder service the name SplunkForwarder and the puppet module expects to just start the service 'splunk' regardless of whether it is a forwarder or the server.
